fendermint/vm/interpreter/src/fvm/state/genesis.rs

```diff
@@ -464,4 +532,19 @@ where
             Stage::Exec(ref mut exec_state) => g(exec_state.state_tree_mut()),
         }
     }
+
+    /// Query the actor state from the state tree under the two different stages.
+    fn get_actor_state<T: de::DeserializeOwned>(&self, actor: ActorID) -> anyhow::Result<T> {
+        let actor_state_cid = match &self.stage {
+            Stage::Tree(s) => s.get_actor(actor)?,
+            Stage::Exec(s) => s.state_tree().get_actor(actor)?,
```

Comment:
I'm not familiar with this stuff, so defer to @aakoshh's feedback for correctness here.

Reply:
It looks correct, however I think it can be implemented on top of `with_state_tree` already:

```rust
let actor_state_cid = self.with_state_tree(
  |s| s.get_actor(actor),
  |s| s.get_actor(actor)
).ok_or_else(|| anyhow!("actor state {actor} not found, is it deployed?"))?
```

Doesn't make a huge difference in this case, but it's why that method exists.

Reply:
In fact it's already used like that [here](https://github.com/consensus-shipyard/ipc/blob/90dfca6ac2bf9915777a015d15e0bbb1ff6fbdd5/fendermint/vm/interpreter/src/fvm/state/genesis.rs#L365)

Reply:
@raulk this pattern is used because the `FvmGenesisState` starts with a `StateTree` but later evolves into having a full `FvmExecState` to actually run EVM contract deployments.

Reply:
that works for me too, when I looked at the function definition, I had the impression `with_state_tree` is mutating the state and it requires `&mut self`. THis way there is no need for `&mut`

Reply:
Yeah, I didn't want two versions, it was already verbose enough and I had access to `&mut` everywhere I needed this.